subsys/bluetooth/mesh/adv_ext.c

```diff
@@ -0,0 +1,290 @@
+/*  Bluetooth Mesh */
+
+/*
+ * Copyright (c) 2018 Nordic Semiconductor ASA
+ * Copyright (c) 2017 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <debug/stack.h>
+
+#include <net/buf.h>
+#include <bluetooth/bluetooth.h>
+#include <bluetooth/mesh.h>
+
+#define BT_DBG_ENABLED IS_ENABLED(CONFIG_BT_MESH_DEBUG_ADV)
+#define LOG_MODULE_NAME bt_mesh_adv_ext
+#include "common/log.h"
+
+#include "host/hci_core.h"
+
+#include "adv.h"
+#include "net.h"
+#include "proxy.h"
+
+/* Convert from ms to 0.625ms units */
+#define ADV_INT_FAST_MS    20
+
+static struct bt_le_adv_param adv_param = {
+	.id = BT_ID_DEFAULT,
+	.interval_min = BT_MESH_ADV_SCAN_UNIT(ADV_INT_FAST_MS),
+	.interval_max = BT_MESH_ADV_SCAN_UNIT(ADV_INT_FAST_MS),
+#if defined(CONFIG_BT_MESH_DEBUG_USE_ID_ADDR)
+	.options = BT_LE_ADV_OPT_USE_IDENTITY;
+#endif
+};
+
+enum {
+	/** Controller is currently advertising */
+	ADV_FLAG_ACTIVE,
+	/** Currently performing proxy advertising */
+	ADV_FLAG_PROXY,
```

Comment:
Is it better to put proxy and adv in different advertising sets?

Reply:
Perhaps, but it's fairly expensive, as memory would have to be allocated for this in the controller as well. The current mechanism works fine IMO, and prevents us from sending more than one advertisement per 20 ms from the mesh stack.